src/device-state.ts

```diff
@@ -128,14 +128,64 @@ function createDeviceStateRouter() {
 			),
 	);
 
-	router.patch('/v1/device/host-config', (req, res) =>
-		hostConfig
-			.patch(req.body)
-			.then(() => res.status(200).send('OK'))
-			.catch((err) =>
-				res.status(503).send(err?.message ?? err ?? 'Unknown error'),
-			),
-	);
+	router.patch('/v1/device/host-config', async (req, res) => {
+		// Validate user input, returning 400 with malformed PATCH body
+		try {
+			if (!req.body.network) {
```

Comment:
I like these new checks, I'm wondering if they should go in the `host-config` module, since they seem very specific to that. The same with the defined constants. What do you think @20k-ultra ?

Reply:
Or, perhaps these v1 routes should be moved to device-api/v1.ts as commented in the [TODO](https://github.com/balena-io/balena-supervisor/blob/2fa8156b31c60d56d07bee28bd6ff5fab8cfece2/src/device-state.ts#L92) in this file? Logically grouping the routes together makes sense, and it would make it easier for devs new to the codebase to get acclimated.

Reply:
We should definitely do that, but not on this PR, I am working on some other PR that touches this codebase and I don't want it to break 😅.

Reply:
For sure! I will open an issue ^^